### PR TITLE
FOR REVIEW ONLY - go-algorand-sdk v1.12.0 beta.1

### PR DESCRIPTION
**Pull request by algojack**
# 1.12.0-beta.1
# Added
- EncodeAddress
- ABI Interaction
- WaitForConfirmation function
# Changed
- ABI Interface

### CODE REVIEW

**Review by algobarb — APPROVED** [review]
LGTM?

**Review by jasonpaulos — APPROVED** [review]
Looks good

**Review by onetechnical — APPROVED** [review]

File: future/waitForConfirmation.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		// Check that the `waitRounds` has not passed
+		if waitRounds > 0 && currentRound > lastRound+waitRounds {
+			err = fmt.Errorf("Wait for transaction id %s timed out", txid)
+			return
+		}
+
+		txInfo, _, err = c.PendingTransactionInformation(txid).Do(ctx, headers...)
+		if err != nil {
+			return
+		}
```
Comment (winder) [inline future/waitForConfirmation.go:34]: This is incorrect, on a load-balanced system `PendingTransactionInformation` may return an error but we should still wait for the given timeout.

You can find a reference implementation here: https://github.com/algorand/go-algorand-internal/issues/730

Comment (algojack) [reply future/waitForConfirmation.go:34]: Talked to Will, he is continuing to resolve this issue with appropriate teams.

Comment (jasonpaulos) [reply future/waitForConfirmation.go:34]: Fixed by #267

**Pull request merged**